lib/benchee/statistics/percentile.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Benchee.Statistics.Percentile do
+  @moduledoc false
+
+  @doc """
+  Calculates the value at the `percentile_number`-th percentile. Think of this as the
+  value below which `percentile_number` percent of the samples lie. For example,
+  if `Benchee.Statistics.Percentile.percentile(samples, 99)` == 123.45,
+  99% of samples are less than 123.45.
+
+  ## Examples
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 100)
+  5.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 150)
+  5.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, 0)
+  1.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 8, -1)
+  1.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 50)
+  3.0
+
+  iex> Benchee.Statistics.Percentile.percentile([5, 3, 4, 5, 1, 3, 1, 3], 75)
+  4.75
+  """
+  @spec percentile(list(number()), integer()) :: float()
+  def percentile(samples, percentile_number) do
+    percentile(samples, length(samples), percentile_number)
+  end
+
+  @spec percentile(list(number()), integer(), integer()) :: float()
+  def percentile(samples, number_of_samples, percentile_number) when percentile_number > 100 do
+    percentile(samples, number_of_samples, 100)
+  end
+
+  def percentile(samples, number_of_samples, percentile_number) when percentile_number < 0 do
+    percentile(samples, number_of_samples, 0)
+  end
+
+  def percentile(samples, number_of_samples, percentile_number) do
+    sorted = Enum.sort(samples)
+    rank = (percentile_number / 100) * max(0, number_of_samples + 1)
```

Comment:
why `number_of_samples + 1` - i.e. why the + 1, shouldn't we be good without it?

Reply:
@PragTob I had the same thought. But it's part of the formula. everything comes out wrong if you just use `number_of_samples`. Imagine you have a list of 10 items, and you're looking for the 50th percentile (median):

currently: `rank = (50 / 100) * 11 = 5.5`
without `+ 1`: `rank = (50 / 100) * 10 = 5`

So, as-is, we look for the value halfway between the 5th and 6th elements (yay, median). If we remove the `+1`, then we look for the value of the 5th element (boo, not the median).

Reply:
makes sense... for the odd case of 9, 1/2 * 9 would also be 4.5 but we are looking for 5. Thanks.

However.... dum dum dum - we are 0 indexed. So in my example above we should be looking at number 4 not number 5? But maybe that is redeemed through the `drop`... weird.

I think it's ok as the median tests I wrote back in the day + your tests are passing. One of these things where I trust the tests more than my brain :D